includes/Modules/Site_Verification.php

```diff
@@ -37,13 +37,62 @@
 final class Site_Verification extends Module implements Module_With_Scopes {
 	use Module_With_Scopes_Trait;
 
+	/**
+	 * Meta site verification type.
+	 */
+	const VERIFICATION_TYPE_META = 'META';
+
+	/**
+	 * File site verification type.
+	 */
+	const VERIFICATION_TYPE_FILE = 'FILE';
+
+	/**
+	 * Callback invoked after serving verification file.
+	 *
+	 * @var \Closure
+	 */
+	private $post_serve_verification_file_callback;
+
 	/**
 	 * Registers functionality through WordPress hooks.
 	 *
 	 * @since 1.0.0
 	 */
 	public function register() {
 		$this->register_scopes_hook();
+
+		add_action(
+			'admin_init',
+			function() {
+				$this->handle_verification_token();
+			}
+		);
+
+		$print_site_verification_meta = function() {
+			$this->print_site_verification_meta();
+		};
+
+		add_action( 'wp_head', $print_site_verification_meta );
+		add_action( 'login_head', $print_site_verification_meta );
+
+		// Exit handled by callback for testability.
+		$this->post_serve_verification_file_callback = function () {
+			exit;
+		};
+		add_action(
+			'init',
```

Comment:
Any reason not to do this on `after_setup_theme`? This fires much earlier and would prevent a lot of unnecessary logic to run (`init` usually is quite heavy).

Reply:
Modules aren't registered until `init`. See `\Google\Site_Kit\Plugin::register`